### PR TITLE
[SDA-7847] Add byo oidc support

### PR DESCRIPTION
**Pull request by gdbranco**

### CODE REVIEW

**Comment by bardielle** [comment]
@gdbranco please update the example and the readme files as well

**Comment by gdbranco** [comment]
@bardielle Is it okay to merge this?

File: examples/byo_oidc_operator_roles_oidc/create_rosa_sts_byo_oidc_operator_roles_oidc.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.20.0"
+    }
+    ocm = {
+      version = ">= 1.0"
```
Comment (bardielle) [inline examples/byo_oidc_operator_roles_oidc/create_rosa_sts_byo_oidc_operator_roles_oidc.tf:8]: please fix it to "0.0.1"

File: examples/byo_oidc_operator_roles_oidc/create_rosa_sts_byo_oidc_operator_roles_oidc.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+module rosa_sts {
+    source = "terraform-redhat/rosa-sts/aws"
+    version = "0.0.1"
+
+    cluster_id = var.cluster_id
+    rh_oidc_provider_thumbprint = var.oidc_thumbprint
+    rh_oidc_provider_url = var.oidc_endpoint_url
+    create_operator_roles = var.create_operator_roles
```
Comment (bardielle) [inline examples/byo_oidc_operator_roles_oidc/create_rosa_sts_byo_oidc_operator_roles_oidc.tf:31]: I don't think we need it... This is a module for operator roles...

File: examples/byo_oidc_operator_roles_oidc/variables.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    type = string
+    default = ""
+}
+
+variable url {
+    type = string
+    default = "https://api.openshift.com"
+}
+
+variable create_operator_roles {
```
Comment (bardielle) [inline examples/byo_oidc_operator_roles_oidc/variables.tf:32]: same here - I think it should be removed

File: examples/byo_oidc_operator_roles_oidc/variables.tf [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    type = bool
+    default = true
+}
+
+variable create_oidc_provider {
+    type = bool
+    default = true
+}
```
Comment (bardielle) [inline examples/byo_oidc_operator_roles_oidc/variables.tf:40]: you didn`t add `is_hosted_cp`

Also, in the basic example..

File: operator_roles/variables.tf [hunk 5]
```diff
@@ -1,6 +1,7 @@
 variable cluster_id {
     description = "cluster ID"
     type = string
+    default = ""
```
Comment (bardielle) [inline operator_roles/variables.tf:4]: did you check if it's working?

File: examples/byo_oidc_operator_roles_oidc/variables.tf (hunk 4 above)
Comment (gdbranco) [reply examples/byo_oidc_operator_roles_oidc/variables.tf:40]: That was for HS support, should be separated. Thanks!

File: operator_roles/variables.tf (hunk 5 above)
Comment (gdbranco) [reply operator_roles/variables.tf:4]: Removing for now as we should check this better later

File: examples/byo_oidc_operator_roles_oidc/create_rosa_sts_byo_oidc_operator_roles_oidc.tf (hunk 2 above)
Comment (gdbranco) [reply examples/byo_oidc_operator_roles_oidc/create_rosa_sts_byo_oidc_operator_roles_oidc.tf:31]: This is needed as we don't necessarily want to create the operator roles as they can be reused. So when using the module the user might want to create the provider and not the operator roles and vice-versa.

**Review by bardielle — APPROVED** [review]

**Pull request merged**